sklearn/cluster/_agglomerative.py

```diff
@@ -747,6 +747,13 @@ class AgglomerativeClustering(ClusterMixin, BaseEstimator):
 
         .. versionadded:: 0.21
 
+    compute_distances : bool, default=False
```

Comment:
I know this was suggested in https://github.com/scikit-learn/scikit-learn/pull/16903#discussion_r426198366

but I still have a doubt between `compute_distance`, `compute_distances` and `return_distance`.  Any preferences @ogrisel ?

Reply:
AFAIK `return_distance` is used for functions or methods that actually return distances.

So I am fine with using `compute_*` for a constructor param.

Between `compute_distance` and `compute_distances`, I am not sure. But since the name of the attribute is `distances_`, I find it ok to use the plural for this case.